pkgs/by-name/sh/shapr3d/package.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  fetchurl,
+  stdenv,
+  undmg,
+}:
+let
+  inherit (stdenv.hostPlatform) system;
+  unsupported = throw "Shapr3D doesn't support ${system}";
+  identifier = "5.910.0.9193";
+  code =
+    {
+      aarch64-darwin = "aCxANPRSe/biEZVL2fHOHt1h7b2vMDwafuaU9NHoy5Q=";
+    }
+    .${system} or unsupported;
+  url =
+    {
+      aarch64-darwin = fetchurl {
+        url = "https://download.shapr3d.com/mac/Shapr3D-${identifier}.dmg";
+        sha256 = code;
+      };
+    }
+    .${system} or unsupported;
+in
+stdenv.mkDerivation {
+  pname = "shapr3d";
+  version = identifier;
+  src = url;
+
+  nativeBuildInputs = [ undmg ];
+
+  sourceRoot = "Shapr3D.app";
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/Applications/Shapr3D.app
+    cp -R . $out/Applications/Shapr3D.app
```

```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/Applications
    cp -R Shapr3D.app $out/Applications
```
`sourceRoot` is more for cases when you need to run a bit more intricate executable, like `make`